### ISSUE TITLE
Provide Specific Prometheus version not respected

### ISSUE THREAD

**Issue by JBaptisme33** [prometheus-operator/prometheus-operator#4550]
**What happened?**

When specified versions is provided into `.Spec.Version`, it seem not be take in account

**Environment**

* Prometheus Operator version:

  tested on `v0.35.0` and `v0.54.0`

* Kubernetes version information:
`1.21`

* Kubernetes cluster kind:
GKE
* Manifests:
Kustomize used. Version requested as follow:
```
- op: replace
  path: /spec/version
  value: v2.31.0
```
```
 k -n monitoring get prometheuses.monitoring.coreos.com             
NAME   VERSION   REPLICAS   AGE
k8s    v2.31.0   1          2y62d
```
```
k -n monitoring get sts prometheus-k8s -o yaml| grep image                      
imagePullPolicy: Always
image: quay.io/prometheus-operator/prometheus-config-reloader:v0.54.0
imagePullPolicy: IfNotPresent
image: quay.io/thanos/thanos:v0.18.0
imagePullPolicy: IfNotPresent
image: quay.io/prometheus-operator/prometheus-config-reloader:v0.54.0
imagePullPolicy: IfNotPresent
```

Prometheus has been rebooted yesterday, so, we are on latest image (2.33.0 instead of 2.31)
![image](https://user-images.githubusercontent.com/59478661/152121052-ced54683-5ac4-4d21-8c51-0f5d9b726d87.png)

**Existing Workaround:**
Fix version with baseimg like this seem to fix issue:
```
- op: replace
  path: /spec/image
  value: quay.io/prometheus/prometheus:v2.31.0
```

What's right way to fix Prom version ? 

Regards,

**Comment by simonpasquier** [comment]
`.spec.version` tells the operator about the version to use when generating the configuration. This should be consistent with the image referenced by `.spec.image` but the operator can't enforce it.
I agree that we should try to reduce the confusion though it can be tricky to get right.

**Comment by simonpasquier** [comment]
fixed by #5171
